### PR TITLE
WT-624: Complete refactor of navbar and nav dropdowns

### PR DESCRIPTION
**Pull request by nateruble**
### Description
This change combines `Navigation.tsx` and `udash_navbar.tsx` into one reusable file (also called `Navigation.tsx` which is in the components directory. This allows a new navbar to be defined by passing the new `Navigation` an array of objects, each of contains the name, location, and symbol for a button on the navbar.

I couldn't find anything about the dropdowns that needed to be fixed. It looks like someone got to them last term and didn't say anything about it.

### Risks
Ensuring that each file that implemented one of the old navbar components now uses the new navbar and that navbar works

### Validation
Walked through the different pages of the site and ensured that navbar shows up and works.

### Issue
[WT-624](https://github.com/orgs/WonderTix/projects/3?pane=issue&itemId=55314811)

### Operating System
Changes made and tested on windows 11

### CODE REVIEW

**Review by henrykaus — COMMENTED** [review]
Everything looks on the right track, just a little bit of feedback. Let me know if you have any questions

File: client/src/components/Navigation.tsx [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+                viewBox="0 0 24 24"
+                stroke="currentColor"
+                strokeWidth={2}
+              >
+                <path
+                  strokeLinecap="round"
+                  strokeLinejoin="round"
+                  d={symbol}
+                />
+              </svg>
```
[the remark this answers is not in the document]
Comment (nateruble) [reply client/src/components/Navigation.tsx]: I think we should make centralizing the svg symbols it's own ticket

[the remark this answers is not in the document]
Comment (henrykaus) [reply client/src/components/Navigation.tsx]: Yeah, that makes sense. It’s a bit more of lift

File: client/src/components/Navigation.tsx [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+          />
+          <div className="absolute z-40 sm:relative w-[14rem] md:w-96 shadow bg-zinc-900 md:hidden transition duration-150 ease-in-out h-full">
+            <div className="flex flex-col justify-between h-full w-full overflow-auto">
+              <div>
+                <div className="flex items-center justify-between px-6">
+                  <div className="h-12  w-full flex flex-col ml-6 items-center">
+                    <button className="flex flex-col  bg-transparent border-transparent justify-center items-center">
+                      <img src={logo} className="mt-[5rem] w-[6rem]" alt="/"
+                        onClick={() => navigate('/')}/>
+                    </button>
```
Comment (henrykaus) [inline client/src/components/Navigation.tsx:139]: Usually when it comes to `onClick`s, they should almost always belong to `button` elements (with rare exceptions such as the `div` above to close the menu), as a button is something that is inherently clickable. So this `onClick` should be a property of the button.

If you're noticing that moving the onClick to the button lets the user click outside the img and still navigate to the homepage, then you could move the `mt-[5rem]` class to that button as well.

Comment (nateruble) [reply client/src/components/Navigation.tsx:139]: I put the onClick on the image because it did not work on the button

**Review by henrykaus — APPROVED** [review]
This looks great, and its nice to now have everything all in one place! Thanks for making all of the changes!

**Review by henrykaus — COMMENTED** [review]
Looks like the merge conflict needs to be resolved, and then I can re-approve.

**Review by henrykaus — APPROVED** [review]
Once the tests pass, feel free to merge!

**Pull request merged**